src/metric.ts

```diff
@@ -56,14 +60,30 @@ export class MetricRegistry {
       }
     }
 
+    for (const [metricName, metrics] of Object.entries(result)) {
+      // Emit each metric name as separate event
+      // Batch the metrics to avoid event emitter overhead
+      // Do not modify the original metrics array.
+      const batchSize = 250
+      let i = 0
+      const metricsBatch = [...metrics]
+      for (i = 0; i < metricsBatch.length; i += batchSize) {
+        this.emit(metricName, {
+          type: metricName,
+          metrics: metricsBatch.slice(i, i + batchSize)
```

Comment:
Why is this batching done? Why not just send the all the metrics off in one event?

Reply:
Good point. I have removed it now, it doesn't make sense to batch the metric specific event anymore. (my initial implementation was to send all metrics and this was a leftover)